### PR TITLE
Respect initial window state

### LINKED ISSUE #1669: glmark2-es2 --fullscreen on XWayland is not fullscreen
One way to reproduce:
```
mir_performance_tests --gtest_filter=GLMark2Xwayland.fullscreen
```

There should be no decoration afaict?

### PR DESCRIPTION
**Pull request by wmww**
Fixes #1669

### CODE REVIEW

File: src/server/frontend_xwayland/xwayland_surface.cpp [hunk 1]
```diff
@@ -643,6 +640,28 @@ void mf::XWaylandSurface::move_resize(uint32_t detail)
...
+{
+    bool nil{false}, *prop_ptr = &nil;
+
+    if (net_wm_state == connection->net_wm_state_hidden)
+        prop_ptr = &minimized;
+    else if (net_wm_state == connection->net_wm_state_maximized_horz) // assume vert is also set
+        prop_ptr = &maximized;
+    else if (net_wm_state == connection->net_wm_state_fullscreen)
+        prop_ptr = &fullscreen;
+
```
Comment (RAOF) [inline src/server/frontend_xwayland/xwayland_surface.cpp:656]: Should we be logging anything when we've asked to apply a change to a state we don't handle?

File: src/server/frontend_xwayland/xwayland_surface.cpp [hunk 2]
```diff
@@ -643,6 +640,28 @@ void mf::XWaylandSurface::move_resize(uint32_t detail)
     }
 }
 
+void mf::XWaylandSurface::WindowState::apply_change(
+    std::shared_ptr<XCBConnection> const& connection,
+    NetWmStateAction action,
+    xcb_atom_t net_wm_state)
+{
+    bool nil{false}, *prop_ptr = &nil;
```
Comment (AlanGriffiths) [inline src/server/frontend_xwayland/xwayland_surface.cpp:648]: I guess it is pre-existing as the code has simply moved, but this isn't the project style.

File: src/server/frontend_xwayland/xwayland_surface.cpp (hunk 1 above)
Comment (AlanGriffiths) [reply src/server/frontend_xwayland/xwayland_surface.cpp:656]: Pre-existing.

File: src/server/frontend_xwayland/xwayland_surface.cpp [hunk 3]
```diff
@@ -643,6 +640,28 @@ void mf::XWaylandSurface::move_resize(uint32_t detail)
...
+    else if (net_wm_state == connection->net_wm_state_maximized_horz) // assume vert is also set
+        prop_ptr = &maximized;
+    else if (net_wm_state == connection->net_wm_state_fullscreen)
+        prop_ptr = &fullscreen;
+
+    switch (action)
+    {
+    case NetWmStateAction::REMOVE:  *prop_ptr = false;      break;
+    case NetWmStateAction::ADD:     *prop_ptr = true;       break;
+    case NetWmStateAction::TOGGLE:  *prop_ptr = !*prop_ptr; break;
```
Comment (AlanGriffiths) [inline src/server/frontend_xwayland/xwayland_surface.cpp:661]: I guess it is pre-existing as the code has simply moved, but this isn't the project style.

**Pull request merged**